wingtips-zipkin2/src/test/java/com/nike/wingtips/zipkin2/util/WingtipsToZipkinSpanConverterDefaultImplTest.java

```diff
@@ -177,6 +178,40 @@ public void convertWingtipsSpanToZipkinSpan_throws_IllegalArgumentException_when
         assertThat(ex).isInstanceOf(IllegalArgumentException.class);
     }
 
+    @DataProvider(value = {
+            "123e4567-e89b-12d3-a456-426655440000  ",  // UUID format (hyphens and also >32 chars)
```

Comment:
Yeah, an unfortunate side effect of wingtips dealing with strings rather than converting to longs internally (and therefore rejecting invalid ID formats) is that people don't pay attention to the format and do whatever they feel like with IDs. This means that we have UUIDs being tossed around in the wild, and even more exotic "IDs" in some cases. 

So as much as I don't like that we have to do this, this kind of sanitization is necessary if we want to integrate with other tools that don't accept any-old-string for IDs.

Brown field is never pretty. :(

Reply:
@longtonthat I'll leave it up to you whether you want to do what Adrian's suggesting. Similar to the `attemptToConvertToLong(...)` logic, you could use `UUID.fromString(...)` to try and convert the original ID to a UUID. If it works, then you could just strip the hyphens and use that. You could only do this when `allow128Bit` is true though.